### ISSUE TITLE
Accessibility in API Client

### ISSUE THREAD

**Issue by VinoManoTR** [scalar/scalar#4051]
### What happens?

1. The Base URL and History button does not get announced by screen reader.
![image](https://github.com/user-attachments/assets/b26a4119-95a0-4312-bbbe-b171f35a762a)

2. When the side panel is expanded the focus currently moves down the list as the user presses tab. But the focus should move into the main area after pressing tab. Also after pressing enter to select any item of side navigation, the focus stays on the selected item, but the focus should move into the main area of the selected item.

3. The accordion labels (Authentication, Cookies, Headers, etc.) should have heading tags \<h2\>. Also the accordion areas are missing their role (role='region'). 

4. The check box in auth dropdown is not accessible from keyboard and doesn't have a role (role='checkbox') 
![image](https://github.com/user-attachments/assets/73cef509-5cdf-4d7a-ace1-046924ac9dfb)

5. Also the combo box name in the same auth section is not announced by screen readers.
![image](https://github.com/user-attachments/assets/7ed582ea-d215-498e-b74e-349a35d862ba)

6. The table in Authentication has the first row set as a table header, but it's the first column that are set as headers. The other tables (cookies, headers, query parameter, etc.) also are missing table headers. Also the enable/disable checkbox on the rows do not identify which row is being enable/disabled on a screen reader.
![image](https://github.com/user-attachments/assets/de636dd0-c6b6-4fb2-b999-e92a7a4052a1)

7. These buttons should be tabs instead of radio buttons
![image](https://github.com/user-attachments/assets/2ad0b09a-45aa-4071-8179-10f115bfb596)

8. The response body section has a horizontal scroll if the lines are too long which are not accessible for keyboard users. 
The request body has a focus trap, where the user can't exit the body using keyboard.

### What did you expect to happen?

All active elements on the test request page should be keyboard accessible.
When the keyboard focus is on an element, there should be a visual indicator of the focus (usually a box highlighting the element).
The screen reader should be able to make an announcement of each element that includes context for the visual label.
Tables need to have a <th> table header to properly work with screen reader.

**Comment by hwkr** [comment]
Hi @VinoManoTR! Thanks for reporting these (and the other a11y) issues :)

Are you able to provide your screen reader setup including:
- Operating System
- Browser Version
- Screenreader software

Most of the testing I've done is on MacOS, Chrome and Voiceover so we're probably missing some stuff for other setups.

**Comment by VinoManoTR** [comment]
Hi, sorry for the late reply!
Most of these were tested on Windows, using Chrome, and NVDA as the screen reader

**Comment by hwkr** [comment]
> Hi, sorry for the late reply! Most of these were tested on Windows, using Chrome, and NVDA as the screen reader

No worries! Thanks for the details :)
